apps/emails-and-messages/src/modules/sendgrid/ui/dangrous-section.tsx

```diff
@@ -0,0 +1,73 @@
+import { Box, Button, Text } from "@saleor/macaw-ui/next";
+import { BoxWithBorder } from "../../../components/box-with-border";
+import { SectionWithDescription } from "../../../components/section-with-description";
+import { defaultPadding } from "../../../components/ui-defaults";
+import { BoxFooter } from "../../../components/box-footer";
+import { SendgridConfiguration } from "../configuration/sendgrid-config-schema";
+import { trpcClient } from "../../trpc/trpc-client";
+import { useDashboardNotification } from "@saleor/apps-shared";
+import { useForm } from "react-hook-form";
+import { SendgridGetConfigurationIdInput } from "../configuration/sendgrid-config-input-schema";
+import { useRouter } from "next/router";
+import { appUrls } from "../../app-configuration/urls";
+import { setBackendErrors } from "../../../lib/set-backend-errors";
+
+interface DangerousSectionProps {
+  configuration: SendgridConfiguration;
+}
+
+export const DangerousSection = ({ configuration }: DangerousSectionProps) => {
```

Comment:
I'd extract generic component (DangerousSection) which will this red box, but inside `modules/sendgrid` I'd call it something like `RemovedSendgridConfigSection`

You can extract it even further like this:

```
import {DangeorousSection} from "UI"
import {RemoveProviderSection} from "modules/providers-configuration"

export const RemoveSendgridProvider = () => {
  const {mutate} = trpc.sendgrid.remove.useMutation() // example

  return <RemoveProviderSection onRemove={mutate} />
}


export const RemoveSmptProvider = () => {
  const {mutate} = trpc.smpt.remove.useMutation() // example

  return <RemoveProviderSection onRemove={mutate} />
}

```

So you share UI and compose only props on top of the UI (that will differ only with endpoint call)

Reply:
Module extracted